components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/cache/OrganizationDetailsCacheEntry.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.organization.management.service.cache;
+
+import java.util.List;
+
+/**
+ * Organization cache entry.
+ */
+public class OrganizationDetailsCacheEntry extends CacheEntry {
+
+    private String orgName;
+    private List<String> ancestorOrganizationIds;
+    private Integer organizationDepthInHierarchy;
```

Comment:
```suggestion
    private int organizationDepthInHierarchy;
```
Any reason not to use primitive here?

Reply:
If no any special reason to have Integer object, correct all usages as well

Reply:
Default value of primitive data type int is 0 and it can not be set to null. Need to keep the value of organizationDepthInHierarchy attribute as null when it is not set in order to identify whether the value is set or not.